code/game/objects/items/holy_weapons.dm

```diff
@@ -1,3 +1,32 @@
+/datum/component/anti_magic
+	var/magic = FALSE
+	var/holy = FALSE
+
+/datum/component/anti_magic/Initialize(_magic = FALSE, _holy = FALSE)
+	magic = _magic
+	holy = _holy
+
+/mob/proc/anti_magic_check(magic = TRUE, holy = FALSE)
+	if(!magic && !holy)
+		return
+	var/list/obj/item/item_list = list()
+	if(isliving(src))
+		var/mob/living/L = src
+		item_list |= L.get_equipped_items(TRUE)
+	item_list |= held_items
+	for(var/obj/O in item_list)
+		GET_COMPONENT_FROM(anti_magic, /datum/component/anti_magic, O)
+		if(!anti_magic)
+			continue
+		if((magic && anti_magic.magic) || (holy && anti_magic.holy))
```

Comment:
do the logic inside the component, best bet is probably to check a signal return flag